### ISSUE TITLE
Use custom environment to run this code

### ISSUE THREAD

**Issue by eric-zqwang** [timoklein/alphazero-gym#14]
Hi, 
I want to use A0 single player method to run on my custom environment.

My state is a picture which has [2, 50, 50] shape. The action is also discrete. I changed the DiscretePolicy self.trunk from the fully connected layer to a CNN. But I'm really confused why it can not work. The policy loss is increased and seems like the agent can not learn anything. 

I'm appreciate if you have any suggestions.

Sincerely

![Screenshot from 2023-08-07 17-08-27](https://github.com/timoklein/alphazero-gym/assets/85941647/7b901c16-4e20-4fc7-8f5e-905332e707ab)

**Comment by timoklein** [comment]
Hard to say from these metrics. They don't provide a lot of information. Here's a couple of things you can try to log to gather more insight into what's actually going wrong:

- Explained variation, something like $\frac{\text{Var}(\hat V (\mathbf s) - V_\theta (\mathbf s))}{\text{Var}(\hat V (\mathbf s))}~.$ This should tell you whether your value function is learning something that make sense.
- Gradient norms. Maybe gradients are exploding? Learning rate is too high?
- Episode length. If it's stuck at a specific value, maybe your policy has collapsed.
- Relative entropy, i.e. current policy entropy divided by $- \log(\frac{1}{|\mathcal A|})~,$ which is the maximum entropy your Categorical policy distribution has. If it stays large, the policy doesn't learn. If it drops to near zero, your policy collapses.

If neither of these give you any insight, there's no other option than to step through the code. Might be a bug in your code or my code.

**Comment by timoklein** [comment]
Closing this due to the lack of a reply. Feel free to reopen it when you have more information/questions.
